pkg/kubelet/cm/dra/manager_test.go

```diff
@@ -0,0 +1,405 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	v1 "k8s.io/api/core/v1"
+	resourcev1alpha2 "k8s.io/api/resource/v1alpha2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/dynamic-resource-allocation/resourceclaim"
+	drapbv1 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/plugin"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+const (
+	driverName      = "test-cdi-device"
+	driverClassName = "test"
+)
+
+var (
+	socketName string
+	stopCh     = make(chan struct{})
+)
+
+type fakeDRADriverGRPCServer struct {
+	drapbv1.UnimplementedNodeServer
+	driverName string
+	mutex      sync.Mutex
+	devices    map[string]string
+}
+
+func (s *fakeDRADriverGRPCServer) NodePrepareResource(ctx context.Context, req *drapbv1.NodePrepareResourceRequest) (*drapbv1.NodePrepareResourceResponse, error) {
+	deviceName := "claim-" + req.ClaimUid
+	result := s.driverName + "/" + driverClassName + "=" + deviceName
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	s.devices[deviceName] = result
+	return &drapbv1.NodePrepareResourceResponse{CdiDevices: []string{result}}, nil
+}
+
+func (s *fakeDRADriverGRPCServer) NodeUnprepareResource(ctx context.Context, req *drapbv1.NodeUnprepareResourceRequest) (*drapbv1.NodeUnprepareResourceResponse, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	delete(s.devices, "claim-"+req.ClaimUid)
+	return &drapbv1.NodeUnprepareResourceResponse{}, nil
+}
+
+func setupFakeDRADriverGRPCServer(t *testing.T) {
+	socketDir, err := os.MkdirTemp("", "dra")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	socketName = filepath.Join(socketDir, "server.sock")
+
+	l, err := net.Listen("unix", socketName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := grpc.NewServer()
+	fakeDRADriverGRPCServer := &fakeDRADriverGRPCServer{
+		mutex:      sync.Mutex{},
+		devices:    make(map[string]string),
+		driverName: driverName,
+	}
+
+	drapbv1.RegisterNodeServer(s, fakeDRADriverGRPCServer)
+
+	go func() {
+		go s.Serve(l)
+		<-stopCh
+		s.GracefulStop()
+	}()
+}
+
+func TestNewManagerImpl(t *testing.T) {
+	kubeClient := fake.NewSimpleClientset()
+	stateFileDirectory := t.TempDir()
+	manager, err := NewManagerImpl(kubeClient, stateFileDirectory)
```

Comment:
no need to use t.TempDir(), just use draManagerStateFileName, I've manually test on local

Reply:
done in 04cfefa673c